src/processTargets/targets/BaseTarget.ts

```diff
@@ -0,0 +1,183 @@
+  protected readonly state: BaseTargetParameters;
+
+  constructor(parameters: BaseTargetParameters) {
+    this.state = {
+      editor: parameters.editor,
+      isReversed: parameters.isReversed,
+      contentRange: parameters.contentRange,
+      position: parameters.position,
+      delimiter: parameters.delimiter,
+      thatTarget: parameters.thatTarget,
+    };
+  }
+
+  get position() {
+    return this.state.position;
+  }
+  get editor() {
+    return this.state.editor;
+  }
+  get isReversed() {
+    return this.state.isReversed;
+  }
+  get contentRemovalRange() {
+    return this.contentRange;
+  }
+  get isLine() {
+    return false;
+  }
+  get isParagraph() {
+    return false;
+  }
+  get isWeak() {
+    return false;
+  }
+
+  get thatTarget(): Target {
+    return this.state.thatTarget != null
+      ? this.state.thatTarget.thatTarget
+      : this;
+  }
+
+  get contentText(): string {
+    return this.editor.document.getText(this.contentRange);
+  }
+
+  get contentSelection(): Selection {
+    return selectionFromRange(this.isReversed, this.contentRange);
+  }
+
+  get contentRange(): Range {
+    switch (this.position) {
+      case "start":
+      case "before":
+        return new Range(
+          this.state.contentRange.start,
+          this.state.contentRange.start
+        );
+      case "end":
+      case "after":
+        return new Range(
+          this.state.contentRange.end,
+          this.state.contentRange.end
+        );
+      default:
+        return this.state.contentRange;
+    }
+  }
+
+  get delimiter(): string | undefined {
+    switch (this.position) {
+      // This it NOT a raw target. Joining with this should be done on empty delimiter.
+      case "start":
+      case "end":
+        return "";
+      default:
+        return this.state.delimiter;
+    }
+  }
+
+  get leadingDelimiterRange() {
+    const { includeDelimitersInRemoval, leadingDelimiterRange } =
+      getTokenDelimiters(this.state.editor, this.state.contentRange);
+    return includeDelimitersInRemoval || this.position != null
+      ? leadingDelimiterRange
+      : undefined;
+  }
+
+  get trailingDelimiterRange() {
+    const { includeDelimitersInRemoval, trailingDelimiterRange } =
+      getTokenDelimiters(this.state.editor, this.state.contentRange);
+    return includeDelimitersInRemoval || this.position != null
+      ? trailingDelimiterRange
+      : undefined;
+  }
+
+  maybeAddDelimiter(text: string): string {
+    return maybeAddDelimiter(text, this.delimiter, this.position);
+  }
+
+  getEditNewContext(isBefore: boolean): EditNewContext {
+    if (this.delimiter === "\n" && !isBefore) {
+      return { type: "command", command: "editor.action.insertLineAfter" };
+    }
+    return {
+      type: "delimiter",
+      delimiter: this.delimiter ?? "",
+    };
+  }
+
+  getRemovalRange(): Range {
+    switch (this.position) {
+      case "before":
+        return this.leadingDelimiterRange ?? this.contentRange;
```

Comment:
How come we fall back to content range here?

Reply:
Is just because the removal action requires a target and in this case the content range will always be empty. This basically says that if we have a leading the limited range remove that or remove an empty range ie do nothing. We should probably make the method return undefined instead.

Reply:
Ah right. So it's a way of returning empty range. Maybe just add a comment for now